module/VuFindTheme/src/VuFindTheme/ResourceContainer.php

```diff
@@ -129,12 +129,101 @@ public function addCss($css)
      */
     public function addJs($js)
     {
-        if (!is_array($js) && !is_a($js, 'Traversable')) {
-            $js = [$js];
+        if ((!is_array($js) && !is_a($js, 'Traversable')) || isset($js['file'])) {
```

Comment:
It seems that addJs handles quite a few different kinds of input. Would it make sense to add a new method for adding the keyed version and make it and this one call an internal method after massaging the input to a common format?

Reply:
The logic has been split up in the meantime, but this affects the addJsEntry function rather than the addJs function itself.